python_modules/libraries/dagster-databricks/dagster_databricks/components/databricks_workspace/component.py

```diff
@@ -0,0 +1,215 @@
+
+        workspace_data = deserialize_value(state_path.read_text(), DatabricksWorkspaceData)
+        jobs_state = workspace_data.jobs
+
+        databricks_assets = []
+
+        for job in jobs_state:
+            job_specs = []
+            task_key_map = {}
+
+            tasks = job.tasks or []
+
+            for task in tasks:
+                current_task_key = getattr(task, "task_key", "unknown")
+                specs = self.get_asset_specs(job_name=job.name, task=task)
+
+                for spec in specs:
+                    job_specs.append(spec)
+                    task_key_map[spec.key] = current_task_key
+
+            if not job_specs:
+                continue
+
+            asset_name = f"databricks_job_{job.job_id}"
+
+            #  Factory to capture job-level variables
+            def make_job_asset_fn(
+                captured_job_id, captured_job_name, captured_specs, captured_key_map
+            ):
+                # Define one multi-asset for the whole job
+                @multi_asset(name=asset_name, specs=captured_specs, can_subset=True)
+                def _job_multi_asset(context: AssetExecutionContext):
+                    """Execute the Databricks job containing these tasks."""
+                    client = self.workspace.get_client()
+
+                    selected_keys = context.selected_asset_keys
+                    context.log.info(
+                        f"Triggering Databricks job {captured_job_id} for keys: {selected_keys}"
+                    )
+
+                    run = client.jobs.run_now(job_id=captured_job_id)
+
+                    if run.run_page_url:
+                        context.log.info(f"Databricks run URL: {run.run_page_url}")
+
+                    client.jobs.wait_get_run_job_terminated_or_skipped(run.run_id)
+
+                    final_run = client.jobs.get_run(run.run_id)
+
+                    state_obj = final_run.state
+                    result_state = state_obj.result_state if state_obj else None
+
+                    if result_state != RunResultState.SUCCESS:
+                        status_str = result_state.value if result_state else "UNKNOWN"
+                        error_msg = (
+                            f"Databricks job {captured_job_id} failed with state: {status_str}. "
+                            f"View run: {run.run_page_url}"
+                        )
+                        context.log.error(error_msg)
+                        raise Exception(error_msg)
+
+                    context.log.info(f"Job finished with state: {result_state.value}")
+
+                    for spec in captured_specs:
+                        if spec.key in selected_keys:
+                            task_key = captured_key_map.get(spec.key)
+                            yield MaterializeResult(
+                                asset_key=spec.key,
+                                metadata={
+                                    "job_id": MetadataValue.int(captured_job_id),
```

Comment:
it might be nice to prefix all of these keys with `dagster-databricks` (e.g. "dagster-databricks/job_id": ...) just to distinguish things a bit.